packages/core/src/components/panel-stack/panelStack.tsx

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+
+import classNames from "classnames";
+import * as React from "react";
+import { CSSTransition, TransitionGroup } from "react-transition-group";
+
+import * as Classes from "../../common/classes";
+import { IProps } from "../../common/props";
+import { PanelView } from "./panelView";
+
+/** Add this interface to your panel components props type. */
+export interface IPanelProps {
+    /**
+     * Call this method to close the current panel. It should be used sparingly as there is always a back button displayed that will call
+     * this method. If there is only one panel on the stack, it cannot be closed and this method will do nothing.
+     */
+    closePanel(): void;
+    /**
+     * Call this method to open a new panel on the top of the stack.
+     * @param component The React component of the new panel.
+     * @param props The props to be passed to the new panel.
+     * @param options Additional options for the new panel.
+     */
+    openPanel<P>(component: React.ComponentClass<P & IPanelProps>, props: P, options: IPanelOptions): void;
+}
+
+export interface IPanel<P = {}> extends IPanelOptions {
+    /** The component being rendered by this panel. */
+    component: React.ComponentClass<P & IPanelProps>;
+    /** The props passed directly to the component prop. */
+    props?: P;
+}
+
+export interface IPanelOptions {
+    /** The title to be displayed above this panel. It is also used as the back button for any panel opened by this panel. */
+    title: string;
+}
+
+export interface IPanelStackProps extends IProps {
+    /** The panel to show when initially rendering the component */
+    initialPanel: IPanel;
```

Comment:
must this be required?

Reply:
@giladgray I think it must. The only way to open a panel (after the initial panel becomes the stack) is from a panel. If there are no more panels, the stack will stay empty, forever. I don't think there's a valid use case for that.

Reply:
oh yeah that makes a lot of sense. probably worth calling that out in some usage docs.